src/hotspot/share/utilities/concurrentHashTable.inline.hpp

```diff
@@ -982,20 +985,21 @@ template <typename CONFIG, MEMFLAGS F>
 template <typename EVALUATE_FUNC>
 inline size_t ConcurrentHashTable<CONFIG, F>::
   delete_check_nodes(Bucket* bucket, EVALUATE_FUNC& eval_f,
-                     size_t num_del, Node** ndel)
+                     size_t num_del, Node** ndel, GrowableArrayCHeap<Node*, F>& extra)
 {
   size_t dels = 0;
   Node* const volatile * rem_n_prev = bucket->first_ptr();
   Node* rem_n = bucket->first();
-  while (rem_n != NULL) {
+  for (dels = 0; rem_n != nullptr; ++dels) {
```

Comment:
`dels` is incremented even when `eval_f(rem_n->value())` is false. This means that the return value may be overcounting, and there will be uninitialized pointer values in the `ndel` array.

Reply:
Thanks! that was bad from my part :-(